### ISSUE TITLE
[PROPOSAL]

### ISSUE THREAD

**Issue by eugenefvdm** [Maatwebsite/Laravel-Nova-Excel#46]
Laravel Nova Excel allows you to gain full control over your resource exports, and you can choose to create your own actions:
https://docs.laravel-excel.com/nova/1.1/exports/customizations.html#full-control

I added an action field and it worked, it's asking me for a value.
https://nova.laravel.com/docs/1.0/actions/defining-actions.html#action-fields

However when I add the handle() method to access the field I get a signature problem and I'm not sure what to do. 

```Return type declaration must be compatible with DownloadExcel->handle(request : \Laravel\Nova\Http\Requests\ActionRequest, exportable : \Laravel\Nova\Actions\Action) : array```

**Comment by GlennM** [comment]
Could you please share a bit more of your code and which versions you're using?
See https://github.com/Maatwebsite/Laravel-Nova-Excel/blob/1.1/.github/ISSUE_TEMPLATE.md if you need any guidance on this.
Thanks!

**Comment by thomassmith10** [comment]
Was a solution provided for this? Thanks.

**Comment by eugenefvdm** [comment]
Hi @thomassmith10 @GlennM asked me to simulate and provide more information but unfortunately I had to move on to other things. So the "request" was closed. Perhaps you can simulate the problem and provide more detail?

**Comment by thomassmith10** [comment]
Thanks @eugenevdm.

I added the below to the Action that extends DownloadExcel;

```
/**
 * Get the fields available on the action.
 *
 * @return array
 */
public function fields()
{
    return [
        Text::make('Subject'),
    ];
}
```

I looked in to the Maatwebsite vendor folder and wasn't able to provide a proper solution so put the following rather hacky solution in place instead; in the map method I simply collect the value via `$subject = $this->request->subject;`.
That works for now, but if an official solution for handling Action fields could be provided for this package that would be great.
